src/v2/components/UserSettings/index.tsx

```diff
@@ -0,0 +1,382 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Query, Mutation, MutationFn } from 'react-apollo'
+import { Form, Field } from 'react-final-form'
+import { FORM_ERROR } from 'final-form'
+
+import Box from 'v2/components/UI/Box'
+import Alert from 'v2/components/UI/Alert'
+import ErrorAlert from 'v2/components/UI/ErrorAlert'
+import LoadingIndicator from 'v2/components/UI/LoadingIndicator'
+import { Input, Textarea, Label, LabelledInput } from 'v2/components/UI/Inputs'
+import { GenericButton as Button } from 'v2/components/UI/GenericButton'
+import Text from 'v2/components/UI/Text'
+import Pulldown from 'v2/components/UI/Pulldown'
+
+import mapErrors from 'v2/util/mapErrors'
+
+import USER_SETTINGS_QUERY from 'v2/components/UserSettings/queries/userSettingsQuery'
+import UPDATE_ACCOUNT_MUTATION from 'v2/components/UserSettings/mutations/updateAccountMutation'
+
+import { MySettings, MySettings_me as Me } from '__generated__/MySettings'
+import {
+  UpdateAccountMutation,
+  UpdateAccountMutationVariables,
+} from '__generated__/UpdateAccountMutation'
+
+interface UserSettingsProps {
+  me: Me
+  updateAccount: MutationFn<
+    UpdateAccountMutation,
+    UpdateAccountMutationVariables
+  >
+}
+
+const Condition = ({ when, is, children }) => (
+  <Field name={when} subscription={{ value: true }}>
+    {({ input: { value } }) => (value === is ? children : null)}
+  </Field>
+)
+
+const InputContainer = styled(LabelledInput)``
+
+const TextInput = styled(Input).attrs({
+  flex: 1,
+  f: 4,
+})``
+
+const Bio = styled(Textarea)``
+
+const Select = styled(Pulldown)``
+
+const UserSettings: React.FC<UserSettingsProps> = ({ me, updateAccount }) => {
+  const isCustom =
+    me.home_path !== '/' &&
+    me.home_path !== '/explore' &&
+    me.home_path !== `/${me.slug}`
+  const homePath = isCustom ? 'custom' : me.home_path
+
+  const handleSubmit = values => {
+    const variables = {
+      ...values,
+      home_path: values.custom_home_path || values.home_path,
+      custom_home_path: undefined,
+      show_nsfw: values.show_nsfw === 'true',
```

Comment:
This is kind of annoying

Reply:
Well you have the control here; don't have the forms output these values?

Reply:
These values are coming from the pulldown which I suppose could handle the parsing of strings to bools but it also seems a little weird.

Reply:
The field should expose correct data; there's nothing to "parse"?

Reply:
This thing probably shouldn't be a pulldown either IMO? (I know it is one currently)

Reply:
Like I'd just write a wrapper for the pulldown that just does generic type coercion or something onChange.

Reply:
Ok will do. What would you suggest if not a pulldown? Radio?

Reply:
Maybe! I think this is just stuck in my head from this morning: https://learnui.design/blog/4-rules-intuitive-ux.html

Reply:
All the 2-choice options could just be horizontally segmented and I could see a vertically segmented button list with the custom home path split into button + input

Reply:
Point taken. Will revisit.

Reply:
But you could of course just ship this for now :)

Reply:
No I think you’re right. It doesn’t seem like all that much of a push to switch. At the very least the two button options. I suspect the home path one might need more designing but I’ll give it a think too

Reply:
No I think you’re right. It doesn’t seem like all that much of a push to switch. At the very least the two button options. I suspect the home path one might need more designing but I’ll give it a think too